src/BlockTypes/ProductQuery.php

```diff
@@ -79,6 +79,60 @@ protected function initialize() {
 		);
 		add_filter( 'rest_product_query', array( $this, 'update_rest_query' ), 10, 2 );
 		add_filter( 'rest_product_collection_params', array( $this, 'extend_rest_query_allowed_params' ), 10, 1 );
+		add_filter( 'render_block_core/query', array( $this, 'add_navigation_id_directive' ), 10, 3 );
+		add_filter( 'render_block_core/query-pagination', array( $this, 'add_navigation_link_directives' ), 10, 3 );
+	}
+
+	/**
+	 * Mark the Product Query as an interactive region so it can be updated
+	 * during client-side navigation.
+	 *
+	 * @param string    $block_content The block content.
+	 * @param array     $block         The full block, including name and attributes.
+	 * @param \WP_Block $instance      The block instance.
+	 */
+	public function add_navigation_id_directive( $block_content, $block, $instance ) {
+		if ( self::is_woocommerce_variation( $block ) ) {
+			// Enqueue the Interactivity API runtime.
+			wp_enqueue_script( 'wc-interactivity' );
+
+			$p = new \WP_HTML_Tag_Processor( $block_content );
+
+			// Add `data-wc-navigation-id to the query block.
+			if ( $p->next_tag( array( 'class_name' => 'wp-block-query' ) ) ) {
+				$p->set_attribute( 'data-wc-interactive', true );
+				$p->set_attribute( 'data-wc-navigation-id', $block['attrs']['queryId'] );
+				$block_content = $p->get_updated_html();
+			}
+		}
+
+		return $block_content;
+	}
+
+	/**
+	 * Add interactive links to all anchors inside the Query Pagination block.
+	 *
+	 * @param string    $block_content The block content.
+	 * @param array     $block         The full block, including name and attributes.
+	 * @param \WP_Block $instance      The block instance.
+	 */
+	public function add_navigation_link_directives( $block_content, $block, $instance ) {
+		if (
+			self::is_woocommerce_variation( $this->parsed_block ) &&
+			$instance->context['queryId'] === $this->parsed_block['attrs']['queryId']
+		) {
+			$p = new \WP_HTML_Tag_Processor( $block_content );
+
+			while ( $p->next_tag( 'a' ) ) {
+				$p->set_attribute(
+					'data-wc-navigation-link',
+					'{"prefetch":true,"scroll":false}'
```

Comment:
I guess it depends on the use-case, but when using the Products block in the Product Catalog template, I think it should default to scrolling to top.

Reply:
I'll make this change as well. 🙂